### PR TITLE
test: pass `dustrelayfee=0` option for tests using dust (instead of `acceptnonstdtxn=1`)

### PR DESCRIPTION
**Pull request by theStack**
By specifying the `dustrelayfee=0` option instead of the more generic `acceptnonstdtxn=1`, we can be more specific about what
part of the transaction is non-standard and can be sure that all other aspects follow the standard policy.
Note that for the test `feature_dbcrash.py`, the UTXO creation at the start of the test has to be split up to several txs to not exceed the tx standard size limit of 100k vbytes

https://github.com/bitcoin/bitcoin/blob/4129c1375430dbfe8dd414868c43fceb3d091fc3/src/policy/policy.h#L26-L27

### CODE REVIEW

File: test/functional/feature_dbcrash.py [hunk 1]
```diff
@@ -211,7 +211,9 @@ def run_test(self):
         self.crashed_on_restart = 0      # Track count of crashes during recovery
 
         # Start by creating a lot of utxos on node3
-        utxo_list = self.wallet.send_self_transfer_multi(from_node=self.nodes[3], num_outputs=5000)['new_utxos']
+        utxo_list = []
+        for _ in range(5):
+            utxo_list.extend(self.wallet.send_self_transfer_multi(from_node=self.nodes[3], num_outputs=1000)['new_utxos'])
```
Comment (MarcoFalke) [inline test/functional/feature_dbcrash.py:216]: why is this needed?

Comment (theStack) [reply test/functional/feature_dbcrash.py:216]: To not exceed MAX_STANDARD_TX_WEIGHT, see PR description.

**Comment by MarcoFalke** [comment]
review ACK 1770be72d5eb47cae565d4ac86de5bc16f94fdd6

**Comment by luke-jr** [comment]
>and can be sure that all other aspects follow the standard policy.

But that's not what is being tested here. This (and the last one like it) are going in the wrong direction. We should be disabling unrelated checks, and only testing the one we're trying to test.

**Comment by MarcoFalke** [comment]
> We should be disabling unrelated checks, and only testing the one we're trying to test.

I am pretty sure this exact topic was discussed previously (years ago), so I am currently not interested in re-hashing it.

If this was something we were doing, it should be disabled globally for all tests, and individual tests that need it could enable it. Though -- obviously -- we are not doing that.

Pull requests with suggestions or reversion, or issues to discuss on a case-by-case basis are welcome.
